### PR TITLE
Np 47873/terms conditions

### PR DESCRIPTION
**Pull request by StigNorland**
I finally found the code.
one reason that it was lost, was that I did a refactoring of all the parameters in this class, and it was too many changes in one go.... still is many...

### CODE REVIEW

**Review by ketilaa — CHANGES_REQUESTED** [review]
As we expect a different behaviour in `UserSelectionUponLoginHandler`, I would like to see a new test `UserSelectionUponLoginHandlerTest#shouldClearUserAttributesAndReturnRestrictedTokenWhenLatestTermsAreNotAccepted()`

**Comment by ketilaa** [comment]
You are aware that @torbjokv is completing the "return token without access right claims if latest version of terms are not approved"? You are still committing code to this branch, so I think you and @torbjokv need to talk.

**Comment by ketilaa** [comment]
Build is failing. Re-request review when fixed.

File: dynamo-client/build.gradle [hunk 1]
```diff
@@ -11,6 +11,7 @@ dependencies {
     implementation libs.aws.auth
     implementation libs.aws.regions
 
+    implementation libs.dynamoDbLocal
```
Comment (ketilaa) [inline dynamo-client/build.gradle:14]: testImplementation?

File: user-access-handlers/src/test/java/no/unit/nva/handlers/HandlerTest.java [hunk 2]
```diff
@@ -59,12 +62,12 @@ private void insertRole(RoleDto role) {
                 databaseService.addRole(role);
             }
         } catch (Exception e) {
-            e.printStackTrace();
+            logger.error("Error inserting role", e);
```
Comment (ketilaa) [inline user-access-handlers/src/test/java/no/unit/nva/handlers/HandlerTest.java:65]: Good. Get rid of code smells like `e.printStackTrace()`

**Review by ketilaa — APPROVED** [review]
Approved given that you fix gradle issue commented.

**Review by torbjokv — APPROVED** [review]
good stuff, lets rethink UserSelectArguments at a later date 👍

**Pull request merged**